### ISSUE TITLE
Inconsistent instructions for advanced installation

### ISSUE THREAD

**Issue by fretboarder** [openshift/openshift-ansible#7010]
#### Description

There are no clear instructions on how to configure the advanced installation with regards to which branch of this repo shall be used.

The README of this repo says:

> We recommend using a release branch

The OpenShift Origin documentation at https://docs.openshift.org/latest/install_config/install/host_preparation.html says:

>  Be sure to stay on the master branch of the openshift-ansible repository when running an advanced installation

From a logical perspective, using the release branch seems to make perfect sense, why would I want to use the master branch at all?

#### Additional question

I'm using the RPM-based installation method. Recently, OpenShift Origin 3.7.1 has be released, so I'm wondering how to set the variables in the inventory properly, so that the 3.7.1 packages are pulled, from the RPM repo as well as from Docker Hub. And which branch do I have to use in that case? My guess would be to use the release-3.7 branch.

**Comment by michaelgugino** [comment]
@fretboarder you should be able to set

openshift_release='v3.7.1'

to get the desired version.  I recommend using the 3.7 release branch for this case.

Thanks for pointing out the doc problem, we'll have to get that patched.

**Comment by fretboarder** [comment]
@michaelgugino 

Obviously, this doesn't work for the RPM install, here's the Ansible failure message:

``
TASK [openshift_version : For an RPM install, abort when the release requested does not match the available version.] ***
fatal: [oo-master-0]: FAILED! => {
    "assertion": "openshift_version.startswith(openshift_release) | bool",
    "changed": false,
    "evaluated_to": false,
    "msg": "You requested openshift_release 3.7.1, which is not matched by\nthe latest OpenShift RPM we detected as origin-3.7.0\non host oo-master-0.\nWe will only install the latest RPMs, so please ensure you are getting the release\nyou expect. You may need to adjust your Ansible inventory, modify the repositories\navailable on the host, or run the appropriate OpenShift upgrade playbook."
}
``

**Comment by michaelgugino** [comment]
@fretboarder that message indicates your system did not detect 3.7.1 rpms.

I only see 3.7.0 here: http://mirror.centos.org/centos/7/paas/x86_64/openshift-origin37/

Origin may have released 3.7.1, but that doesn't necessarily mean that each distro has provided packages for that new release.

It's easier to stay closer to origin's release cycle if you switch to containerized deployments.  The image tags are published in docker hub as soon as a release is made: https://hub.docker.com/r/openshift/origin/tags/

(Note, 3.9 is 'rolling' at the moment, and has not been released yet; it's still testing).

**Comment by fretboarder** [comment]
@michaelgugino 

Using the containerized installation on CentOS doesn't work either. Yes, I'm using the release-3.7 branch.

In my inventory file I have
```
containerized=true
openshift_deployment_type=origin
openshift_release=v3.7.1
```

The installer fails with:
```
TASK [openshift_version : Fail if rpm version and docker image version are different] ***********************
fatal: [oo-master-0]: FAILED! => {"changed": false, "msg": "OCP rpm version 3.7.0 is different from OCP image version 3.7.1"}
```

By the way: According to https://docs.openshift.org/latest/install_config/install/rpm_vs_containerized.html
the default installation method on CentOS is RPM based.

The configuration that works for v3.7.0 is:
```
containerized=false
openshift_deployment_type=origin
openshift_release=v3.7
```

> Origin may have released 3.7.1, but that doesn't necessarily mean that each distro has provided packages for that new release.

Please, can you explain how/why (not)/when the RPM packages become updated? To me it seems that the Origin installation methods / versions / descriptions are a bit of a (horrible) mess.
